### PR TITLE
add: ability to set additional users for a cloudSQL instance database

### PR DESCRIPTION
**Pull request by ybelMekk**
ref: https://github.com/nais/naiserator/issues/226

### CODE REVIEW

File: pkg/resourcecreator/resourcecreator.go [hunk 1]
```diff
@@ -167,8 +167,15 @@ func Create(app *nais_io_v1alpha1.Application, resourceOptions ResourceOptions)
 				if err != nil {
 					return nil, fmt.Errorf("unable to assign sql password: %s", err)
 				}
-				sqlUser := GoogleSqlUser(app, instance, secretKeyRefEnvName, sqlInstance.CascadingDelete, resourceOptions.GoogleTeamProjectId)
+
+				sqlUser := GoogleSqlUser(app, instance, secretKeyRefEnvName, sqlInstance.CascadingDelete, resourceOptions.GoogleTeamProjectId, instance.Name)
```
Comment (ambientsound) [inline pkg/resourcecreator/resourcecreator.go:171]: Better to merge the standard user with the additional users array, to have just one loop creating all the users.

File: pkg/resourcecreator/resourcecreator_test.go [hunk 2]
```diff
@@ -327,6 +327,11 @@ func TestCreate(t *testing.T) {
 						Name: dbName,
 					},
 				},
+				AdditionalUsers: []nais_io_v1alpha1.AdditionalUser{
+					{
+						Name: fmt.Sprintf("%s-%s", instanceName, "read"),
+					},
+				},
```
Comment (ambientsound) [inline pkg/resourcecreator/resourcecreator_test.go:334]: Please avoid adding more tests to resourcecreator_test.go, rather just use the golden file tests only.

File: pkg/resourcecreator/testdata/gcp_database_custom_env.yaml [hunk 3]
```diff
@@ -90,5 +90,27 @@ tests:
...
+              cnrm.cloud.google.com/project-id: team-project-id
+          spec:
+            instanceRef:
+              name: myapplication
+            host: ""
+            password:
+              valueFrom:
+                secretKeyRef:
+                    name: google-sql-myapplication
+                    key: PWNED_PASSWORD
```
Comment (ambientsound) [inline pkg/resourcecreator/testdata/gcp_database_custom_env.yaml:114]: Same password for all users? I think they should have separate passwords.

**Review by ambientsound — CHANGES_REQUESTED** [review]
Looking good, but needs a few fixes

File: pkg/resourcecreator/resourcecreator_test.go (hunk 2 above)
Comment (ybelMekk) [reply pkg/resourcecreator/resourcecreator_test.go:334]: good ide!

File: pkg/resourcecreator/resourcecreator.go (hunk 1 above)
Comment (ybelMekk) [reply pkg/resourcecreator/resourcecreator.go]: good ide, makes code cleaner

File: pkg/resourcecreator/testdata/gcp_database_custom_env.yaml (hunk 3 above)
Comment (ybelMekk) [reply pkg/resourcecreator/testdata/gcp_database_custom_env.yaml:114]: doh. rookie. fixing.

File: pkg/resourcecreator/googlesqluser.go [hunk 4]
```diff
@@ -20,87 +21,169 @@ const (
...
 	}
-	return fmt.Sprintf("NAIS_DATABASE_%s_%s", googleSQLDatabaseCase(instanceName), googleSQLDatabaseCase(db.Name))
 }
 
-func firstKeyWithSuffix(vars map[string]string, suffix string) (string, error) {
+func (in GoogleSqlUser) KeyWithSuffixMatchingUser(vars map[string]string, suffix string) (string, error) {
 	for k := range vars {
 		if strings.HasSuffix(k, suffix) {
-			return k, nil
+			println(k)
```
Comment (mortenlj) [inline pkg/resourcecreator/googlesqluser.go]: Forgotten debugging?

Comment (ybelMekk) [reply pkg/resourcecreator/googlesqluser.go]: Should be gone :D

**Pull request merged**